packages/blade-mcp/src/tools/getBladeComponentDocs.ts

```diff
@@ -49,13 +81,34 @@ const getBladeComponentDocsToolCallback: ToolCallback<typeof getBladeComponentDo
     });
   }
 
-  const createOrUpdateCursorRule = shouldCreateOrUpdateCursorRule(
-    cursorRuleVersion,
-    clientName,
-    currentProjectRootDirectory,
-  );
-  if (createOrUpdateCursorRule) {
-    return createOrUpdateCursorRule;
+  // Skip cursor rule checks if skipLocalCursorRuleChecks is true (for HTTP transport)
+  if (!skipLocalCursorRuleChecks && currentProjectRootDirectory) {
+    const ruleFilePath = join(currentProjectRootDirectory, CONSUMER_CURSOR_RULES_RELATIVE_PATH);
+
+    if (!existsSync(ruleFilePath)) {
+      return handleError({
+        toolName: getBladeComponentDocsToolName,
+        mcpErrorMessage: `Cursor rules do not exist. Call \`${createBladeCursorRulesToolName}\` first.`,
+      });
+    }
+
+    if (hasOutDatedRules(ruleFilePath)) {
+      return handleError({
+        toolName: getBladeComponentDocsToolName,
+        mcpErrorMessage: `Cursor rules are outdated. Call \`${createBladeCursorRulesToolName}\` first to update cursor rules`,
+      });
+    }
+  }
```

Comment:
We can refactor this a bit / move it to a separate function instead of duplicating everywhere.